src/routes/dictionaries/+page.svelte

```diff
@@ -30,16 +30,31 @@
   let createDictionaryError: string | null = null;
   let creatingDictionary: boolean = false;
   let files: FileList;
+  let file: File;
+  let isSequenceAdaptation: boolean = false;
+  let sequenceAdaptationName: string;
 
   $: hasCreatePermission = featurePermissions.commandDictionary.canCreate(data.user);
   $: createButtonDisabled = !files;
+  $: {
+    if (files) {
+      file = files[0];
 
-  async function uploadDictionaryOrAdaptation(files: FileList) {
+      isSequenceAdaptation = file.type === 'application/x-javascript';
```

Comment:
I'm trying to upload an `adaptation.js` file but it looks like the type is `text/javascript` instead of `application/x-javascript`. Any reason not to just check for a `.js` extension instead?

Reply:
The check for `file.type` was just cleaner, I can change it to check against the extension instead.